snow/engine/snowman/engine_test.go

```diff
@@ -520,55 +521,55 @@ func TestEngineRespondsToGetRequest(t *testing.T) {
 func TestEnginePushQuery(t *testing.T) {
 	require := require.New(t)
 
-	vdr, _, sender, vm, te := setup(t, DefaultConfig(t))
+	conf := DefaultConfig(t)
+	vdr, _, sender, vm, te := setup(t, conf)
+
+	vm.LastAcceptedF = snowmantest.MakeLastAcceptedBlockF(
+		[]*snowmantest.Block{{HeightV: 41}},
+	)
+	vm.GetBlockF = func(_ context.Context, _ ids.ID) (snowman.Block, error) {
+		return &snowmantest.Block{HeightV: 41}, nil
+	}
 
 	sender.Default(true)
 
 	blk := snowmantest.BuildChild(snowmantest.Genesis)
 
-	vm.ParseBlockF = func(_ context.Context, b []byte) (snowman.Block, error) {
-		if bytes.Equal(b, blk.Bytes()) {
-			return blk, nil
-		}
-		return nil, errUnknownBytes
+	snowCtx := snowtest.Context(t, snowtest.CChainID)
+	ctx := snowtest.ConsensusContext(snowCtx)
+
+	params := snowball.Parameters{
+		K:                     1,
+		AlphaPreference:       1,
+		AlphaConfidence:       1,
+		Beta:                  3,
+		ConcurrentRepolls:     1,
+		OptimalProcessing:     1,
+		MaxOutstandingItems:   1,
+		MaxItemProcessingTime: 1,
 	}
 
-	vm.GetBlockF = func(_ context.Context, blkID ids.ID) (snowman.Block, error) {
-		switch blkID {
-		case snowmantest.GenesisID:
-			return snowmantest.Genesis, nil
-		case blk.ID():
-			return blk, nil
-		default:
-			return nil, errUnknownBlock
-		}
+	require.NoError(te.Consensus.Initialize(ctx, params, snowmantest.GenesisID, 41, time.Now()))
+
+	vm.ParseBlockF = func(_ context.Context, _ []byte) (snowman.Block, error) {
+		return &snowmantest.Block{HeightV: 41}, nil
```

Comment:
Why are we overriding `ParseBlockF` to return a mocked block w/ height 41 instead of the genesis' child block as in the previous version of this test?

Reply:
I modified the test because I thought it would be less trivial than it is now. I reverted it.